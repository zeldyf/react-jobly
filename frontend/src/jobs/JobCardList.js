import React from 'react';
import JobCard from './JobCard';

/** Show list of job cards.
 *
 * Used by both JobList and CompanyDetail to list jobs.
 *
 * JobList -> JobCardList -> JobCard
 * CompanyDetail -> JobCardList -> JobCard
 *
 */

const JobCardList = ({ jobs }) => {
  console.debug('JobCardList', 'jobs=', jobs);

  return (
    <div className="JobCardList">
      {jobs.map(job => (
        <JobCard
          key={job.id}
          id={job.id}
          title={job.title}
          salary={job.salary}
          equity={job.equity}
          companyName={job.companyName}
        />
      ))}
    </div>
  );
};

export default JobCardList;
