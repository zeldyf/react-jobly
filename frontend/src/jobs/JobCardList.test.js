import React from 'react';
import { render } from '@testing-library/react';
import UserContext from '../UserContext';
import JobCardList from './JobCardList';

describe('JobCardList', () => {
  const jobs = [
    {
      id: 1,
      title: 'Software Engineer',
      salary: '120000',
      equity: '0.01',
      companyName: 'Acme',
    },
    {
      id: 2,
      title: 'Product Manager',
      salary: '150000',
      equity: '0.005',
      companyName: 'Globex',
    },
  ];

  test('renders without errors', () => {
    render(
      <UserContext.Provider value={{ hasAppliedToJob: jest.fn(), applyToJob: jest.fn() }}>
        <JobCardList jobs={jobs} />
      </UserContext.Provider>
    );
  });

  test('matches snapshot', () => {
    const { container } = render(
      <UserContext.Provider value={{ hasAppliedToJob: jest.fn(), applyToJob: jest.fn() }}>
        <JobCardList jobs={jobs} />
      </UserContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
