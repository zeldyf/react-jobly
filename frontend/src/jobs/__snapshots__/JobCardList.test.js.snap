// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCardList matches snapshot 1`] = `
<div>
  <div
    class="JobCardList"
  >
    <div
      class="JobCard mb-3 card"
    >
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          Software Engineer
        </h5>
        <h6>
          Acme
        </h6>
        <p
          class="card-text"
        >
          Salary: 
          120000
        </p>
        <p
          class="card-text"
        >
          Equity: 
          0.01
        </p>
        <button
          class="font-weight-bold text-uppercase float-right btn btn-danger"
          type="button"
        >
          Apply
        </button>
      </div>
    </div>
    <div
      class="JobCard mb-3 card"
    >
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          Product Manager
        </h5>
        <h6>
          Globex
        </h6>
        <p
          class="card-text"
        >
          Salary: 
          150000
        </p>
        <p
          class="card-text"
        >
          Equity: 
          0.005
        </p>
        <button
          class="font-weight-bold text-uppercase float-right btn btn-danger"
          type="button"
        >
          Apply
        </button>
      </div>
    </div>
  </div>
</div>
`;
